src/main/java/org/reaktivity/nukleus/kafka/internal/stream/MessageDispatcher.java

```diff
@@ -21,8 +21,26 @@
 
 public interface MessageDispatcher
 {
+    byte FLAGS_MATCHED = 0x01;
+    byte FLAGS_DELIVERED = 0x02 | FLAGS_MATCHED;
+    byte FLAGS_BLOCKED = 0x04 | FLAGS_MATCHED;
 
-    int dispatch(
+    static boolean matched(byte result)
+    {
+        return (result & FLAGS_MATCHED) > 0;
+    }
+
+    static boolean delivered(byte result)
+    {
+        return (result & FLAGS_DELIVERED) > 0;
+    }
+
+    static boolean blocked(byte result)
+    {
+        return (result & FLAGS_BLOCKED) > 0;
+    }
+
+    byte dispatch(
```

Comment:
Suggest switching all these `byte` types to `int` since there are no `byte` type math instructions in the JVM bytecode so they get cast to and from `int` anyway.

Reply:
done